pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -183,16 +183,35 @@ func buildLogsQuery(panelType v3.PanelType, start, end, step int64, mq *v3.Build
 		having = " having " + having
 	}
 
-	queryTmpl :=
-		"SELECT toStartOfInterval(fromUnixTimestamp64Nano(timestamp), INTERVAL %d SECOND) AS ts" + selectLabels +
-			", %s as value " +
-			"from signoz_logs.distributed_logs " +
-			"where " + timeFilter + "%s " +
-			"group by %s%s " +
-			"order by %s"
-
-	groupBy := groupByAttributeKeyTags(mq.GroupBy...)
+	var queryTmpl string
+
+	if panelType == v3.PanelTypeTable {
+		queryTmpl =
+			"SELECT now() as ts" + selectLabels +
```

Comment:
Can you check once if `now()` is enough with multiple queries in the formula join, or should you round it with something like `toStartOfHour`? More clearly, will now() be the same in both the subqueries and give the final result since we match on exact ts?

Reply:
So I created two queries A and B and formula A + B. The value of now() in them is same. here is the result

```
  {
      "status": "success",
      "data": {
          "resultType": "",
          "result": [
              {
                  "queryName": "A",
                  "series": [
                      {
                          "labels": {
                              "method": "DELETE",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "5"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "GET",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "3"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "POST",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "5"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "PUT",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "7"
                              }
                          ]
                      }
                  ],
                  "list": null
              },
              {
                  "queryName": "B",
                  "series": [
                      {
                          "labels": {
                              "method": "PUT",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "7"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "POST",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "5"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "GET",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "3"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "DELETE",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "5"
                              }
                          ]
                      }
                  ],
                  "list": null
              },
              {
                  "queryName": "F1",
                  "series": [
                      {
                          "labels": {
                              "method": "POST",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "10"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "PUT",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "14"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "DELETE",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "10"
                              }
                          ]
                      },
                      {
                          "labels": {
                              "method": "GET",
                              "severity_number": "13"
                          },
                          "values": [
                              {
                                  "timestamp": 1689152847000,
                                  "value": "6"
                              }
                          ]
                      }
                  ],
                  "list": null
              }
          ]
      }
  }
```